tx-pool/src/component/container.rs

```diff
@@ -179,6 +183,18 @@ impl SortedTxMap {
         }
         // update ancestor_fields
         self.update_ancestors_stat_for_entry(&mut entry, &parents);
+
+        if entry.ancestors_count > self.max_ancestors_count {
+            return Err(SubmitTxError::ExceededMaximumAncestorsCount);
+        }
+
+        // check duplicate tx
+        let removed_entry = if self.contains_key(&short_id) {
+            self.remove_entry(&short_id)
+        } else {
+            None
+        };
```

Comment:
Why do you move this part of the code?

Reply:
If ancestors reach the limit, we should just return an error and not remove the conflict tx. Other wise we should go to the old logic.